content/en/blog/_posts/2021-07-21-memory-manager-beta.md

```diff
@@ -0,0 +1,106 @@
+3. The **MemoryManager** calculates all possible NUMA nodes combinations for every container inside the pod and returns hints to the **TopologyManager**.
+4. The **TopologyManager** calls to `Allocate()` for every hint provider including the **MemoryManager**.
+5. The **MemoryManager** allocates the memory under the state according to the hint that the **TopologyManager** chose.
+
+During the creation phase:
+
+1. The **Kubelet** calls `PreCreateContainer()`.
+2. The **MemoryManager** get the NUMA nodes where it allocated the memory for the container and returns it to **Kubelet**.
+3. The **Kubelet** creates the container via CRI with the spec updated according to the **MemoryManager** information.
+
+### Let's talk about the configuration
+
+By default, the **MemoryManager** runs with the `None` policy, meaning it will just relax and will not do anything. To enable the **MemoryManager**, you should provide two **Kubelet** flags:
+
+- `--memory-manager-policy=Static`
+- `--reserved-memory="<numaNodeID>:<resourceName>=<quantity>"`
+
+It is clear what value you should set for the `--memory-manager-policy` parameter. But it does not 100% clear how you should configure the `--reserved-memory` parameter. To configure it correctly, you should follow two main rules:
+
+- The amount of reserved memory for the `memory` resource must be greater than zero.
+- The amount of reserved memory for the resource type must be equal to [NodeAllocatable](https://kubernetes.io/docs/tasks/administer-cluster/reserve-compute-resources/) (`kube-reserved + system-reserved + eviction-hard`) for the resource. You can read more about 
+
+![ReservedMemory
+](/images/blog/2021-07-21-memory-manager-beta/ReservedMemory.png "ReservedMemory")
+
+## Current limitations
+
+The 1.22 release and promotion to the beta brings along some enhancements and fixes, but the **MemoryManager** still has several limitations.
+
+### Single vs Cross NUMA node allocation
+
+The NUMA node can not have both single and cross NUMA node allocations. When the container memory is pinned to two or more NUMA nodes, we can not know from which NUMA node the container will consume the memory.
+
+![SingleCrossNUMAAllocation
+](/images/blog/2021-07-21-memory-manager-beta/SingleCrossNUMAAllocation.png "SingleCrossNUMAAllocation")
+
+1. The `container1` started on the NUMA node 0 and requests *5Gi* of the memory but currently is consuming only *3Gi* of the memory.
+2. The `container2` the memory request is *10Gi*, and none single NUMA node can satisfy it.
+3. The `container2` consumes *3.5Gi* of the memory from the NUMA node 0, but once the `container1` will require more memory, it will not have it, and the kernel will kill one of the containers with the **OOM** error.
+
+To prevent such issues, the **MemoryManager** will fail the admission of the `container2` until the machine, will have two NUMA nodes without a single NUMA node allocation.
```

Comment:
I'm not sure that I understand this sentence. Do you mean this?: ..."until the machine has two NUMA nodes without a single NUMA node allocation."

Reply:
absolutely, thanks for the comment.